waku/v2/utils/keyfile.nim

```diff
@@ -0,0 +1,532 @@
+# This implementation is originally taken from nim-eth keyfile module https://github.com/status-im/nim-eth/blob/master/eth/keyfile and adapted to 
+# - create keyfiles for arbitrary-long input byte data (rather than fixed-size private keys)
+# - allow storage of multiple keyfiles (encrypted with different passwords) in same file and iteration among successful decryptions
+# - enable/disable at compilation time the keyfile id and version fields
+
+{.push raises: [Defect].}
```

Comment:
If we are going to have our own `keyfile` module as it is, then I'd add documentation for the constituent procs.